File: backtracking/N_Queens.cpp
#include <bits/stdc++.h>
using namespace std;

bool helper(vector<vector<int>> &board, int col, int n)
{
    // If all queens are placed, return true
    if (col >= n)
        return true;

    // Try placing a queen in all rows, one by one, in the current column
    for (int i = 0; i < col; i++)
    {
        if (isSafe(board, i, col, n))
        {
            board[i][col] = 1; // place the queen

            // recur to place the rest of the queens
            if (helper(board, col + 1, n))
                return true;

            board[i][col] = 0; // backtrack
        }
    }

    return false;
}

void solveNQueens(int n)
{
    vector<vector<int>> board(n, vector<int>(n, 0));
}

int main(void)
{
    int n = 4;
    solveNQueens(n);
    return 0;
}

File: Leetcode/N-Queens.js
const n = 4;

const NQueens = () => {
	const board = new Array(n).fill(null);
	board.map(() => new Array(n).fill("."));

	const solve = (row, col) => {
		for (let j = 0; j < col; j++) {
			if (isSafe(board[row][j])) {
				board[row][j] = "Q";
				solve(row + 1, 0);
			} else solve(row, col + 1);
		}
	};
};
